pipeline_dp/pipeline_operations.py

```diff
@@ -38,6 +38,21 @@ def group_by_key(self, col, stage_name: str):
     def filter(self, col, fn, stage_name: str):
         pass
 
+    @abc.abstractmethod
+    def filter_by_key(self, col, public_partitions, stage_name: str):
+        """Filter out the partitions that are not not meant to be public.
+
+        Args:
+          col: input collection
+          public_partitions: collection of public partition keys
```

Comment:
Please add '.' in the end of the comments

Reply:
Done.